lib/generic/falsy.ts

```diff
@@ -0,0 +1 @@
+export type Falsy = 0 | '' | null | undefined | false;
```

Comment:
Do we want to include -0, 0n and NaN as well? As per https://developer.mozilla.org/en-US/docs/Glossary/Falsy

Reply:
Seems that NaN is actually no type (I think it is a number actually)

Reply:
Yea `NaN` doesn't have a constant type in TS and is instead declared as a `number`, despite there being a float32 value specifically for NaN (unsure if JS actually implements this) :/.